joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/typerelations/PointerAssignmentPass.scala

```diff
@@ -0,0 +1,143 @@
+package io.joern.x2cpg.passes.typerelations
+
+import io.shiftleft.codepropertygraph.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes._
+import io.shiftleft.codepropertygraph.generated.{EdgeTypes, Operators}
+import io.shiftleft.passes.ForkJoinParallelCpgPass
+import io.shiftleft.semanticcpg.language._
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+/** Creates an intraprocedural pointer assignment graph. This provides may point to information for each identifier or
+  * field identifier.
+  */
+class PointerAssignmentPass(cpg: Cpg) extends ForkJoinParallelCpgPass[Method](cpg) {
+
+  override def generateParts(): Array[Method] = cpg.method.toArray
+
+  override def runOnPart(builder: DiffGraphBuilder, method: Method): Unit = {
+    val allocSites = method.ast.isCall.name(Operators.alloc, Operators.arrayInitializer).l
+    if (allocSites.nonEmpty) {
+      val assignmentGraph = buildAssignmentGraph(method)
+      val pointsToGraph   = buildPointsToGraph(assignmentGraph)
+      // All references to nodes in the method body should now have their
+      // "may point to" edge added to an allocation site if it is in the map
+      method.ast
```

Comment:
cc @ml86 -- I vaguely remember that some frontends do the terrible thing that they emit inner methods below the AST of the outer method. Is this still correct?

If so, then we need to go via the contains-edge here (the corresponding pass takes that possibility into account)